File: test/unit-test/UriMatchFacts.cs
using System.Net;
using HttpClientMock;
using Xunit;

namespace UnitTest
{
    public class UriMatchFacts
    {
        [Fact]
        public void should_match_url_contains_question_mark()
        {
            var serverBuilder = new MockedHttpClientBuilder();
            serverBuilder
                .WhenGet("/staff?employeeId=Staff0001")
                .Respond(HttpStatusCode.InternalServerError);

            const string baseAddress = "http://localhost:1122";
            using (var httpClient = serverBuilder.Build(baseAddress))
            {
                const string requestUri = "http://localhost:1122/staff?employeeId=Staff0001";
                var response = httpClient.GetAsync(requestUri).Result;
                Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
            }
        }
    }
}

File: test/unit-test/UriMatchFacts.fs
namespace UnitTest

open System.Net
open HttpClientMock
open Xunit
open FsUnit.Xunit

module UriMatchFacts = 

    [<Fact>]
    let ``should match url contains question mark``() =
        let builder = MockedHttpClientBuilder()
        builder.WhenGet("/staff?employeeId=Staff0001")
            .Respond(HttpStatusCode.InternalServerError) |> ignore

        use httpClient = builder.Build("http://localhost:1122")
        async {
            let! response = httpClient.GetAsync("http://localhost:1122/staff?employeeId=Staff0001") |> Async.AwaitTask
            response.StatusCode |> should equal HttpStatusCode.InternalServerError
        } |> Async.RunSynchronously
